### PR TITLE
Fix #1733, Move global count into test global struct.

### LINKED ISSUE #1730: Explicitly document/test file offset behavior in relation to FS file read/write APIs
**Is your feature request related to a problem? Please describe.**
It's not clear from the API where the file offset is left after using any of the FS APIs that access the file:
CFE_FS_ReadHeader
CFE_FS_WriteHeader
CFE_FS_SetTimestamp

**Describe the solution you'd like**
Right now, at least some apps in the wild depend on current behavior so preferred fix for Caelum is just to document current behavior (success and error).  Error can be undefined, but success should be consistent/explicit.

Also worth adding a functional check to confirm this behavior (if it's not already checked).

**Describe alternatives you've considered**
Could explicitly be undefined, or change behavior to reset to the start of the file but that would ripple through all the apps.

**Additional context**
From an app code review, currently undocumented behavior means apps should all lseek but rather than change every app it's likely better to document the current behavior explicitly in the API. 

**Requester Info**
Jacob Hageman - NASA/GSFC, ping @ejtimmon

### LINKED ISSUE #1733: Move "count" global to test global struct
**Describe the bug**
Should not have random global variables around, particularly variables with simple/common names, as it may silently overwrite an existing variable (i.e. if any other app/lib makes a global called "count" some OS's will just happily bind to the existing variable, not make a separate instance, when dynamic refs are resolved after loading).

https://github.com/nasa/cFE/blob/cc8c9a1d79dfdf6c098f18431d852a9e8e28592e/modules/cfe_testcase/src/es_task_test.c#L36

**Expected behavior**
Now that we have a proper global struct, `CFE_FT_Global_t` - This should be moved into it.

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

### PR DESCRIPTION
**Pull request by zanzaben**
**Describe the contribution**
Fixes #1733 
Move global count into the test global struct

**Testing performed**
Build and run unit test

**Expected behavior changes**
No impact to behavior

**System(s) tested on**
Ubuntu 20.04

**Contributor Info - All information REQUIRED for consideration of pull request**
Alex Campbell GSFC

### CODE REVIEW

File: modules/cfe_testcase/src/cfe_test.h [hunk 1]
```diff
@@ -46,6 +46,7 @@
 typedef struct
 {
     CFE_FS_FileWriteMetaData_t FuncTestState;
+    int                        count;
```
Comment (skliper) [inline modules/cfe_testcase/src/cfe_test.h:49]: Recommend either a more useful name or add a comment as to what the intent for this variable is (is it intended as a generic utility global or specifically for use in the es task test?) another improvement would be consistent case (camel).   Helpful also to have `extern CFE_FT_Global_t CFE_FT_Global` in the header, and instantiate it once.

File: modules/cfe_testcase/src/es_task_test.c [hunk 2]
```diff
@@ -33,23 +33,23 @@
 
 #include "cfe_test.h"
 
-uint32 count = 0;
+CFE_FT_Global_t CFE_FT_Global;
```
Comment (skliper) [inline modules/cfe_testcase/src/es_task_test.c:36]: See comment above (define an extern in the header).

File: modules/cfe_testcase/src/es_task_test.c [hunk 3]
```diff
@@ -135,24 +136,24 @@ void TestChildTaskDelete(void)
     size_t                     StackSize    = CFE_PLATFORM_ES_PERF_CHILD_STACK_SIZE;
     CFE_ES_TaskPriority_Atom_t Priority     = CFE_PLATFORM_ES_PERF_CHILD_PRIORITY;
     uint32                     Flags        = 0;
-    count                                   = 0;
+    CFE_FT_Global.count                     = 0;
     int ExpectedCount                       = 5;
```
Comment (skliper) [inline modules/cfe_testcase/src/es_task_test.c:140]: declarations first (coding standard)

File: modules/cfe_testcase/src/es_task_test.c [hunk 4]
```diff
@@ -67,13 +67,14 @@ void TestCreateChild(void)
     CFE_ES_TaskPriority_Atom_t Priority      = CFE_PLATFORM_ES_PERF_CHILD_PRIORITY;
     uint32                     Flags         = 0;
     int                        ExpectedCount = 5;
+    CFE_FT_Global.count                      = 0;
```
Comment (skliper) [inline modules/cfe_testcase/src/es_task_test.c:70]: Add a gap between declarations and other code for readability.

**Review by skliper — CHANGES_REQUESTED** [review]

**Review by skliper — APPROVED** [review]

**Comment by astrogeco** [comment]
[CCB:2021-08-04](https://github.com/nasa/cFS/wiki/Community-CCB-Agenda:-August-04-2021) APPROVED

**Pull request merged**